From 92107214766699fa1fd561dd547f76ca338b21cf Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:47:43 -0400
Subject: [pci] hotplug/apci: Use list_for_each_entry() for bus traversal

Message-id: <20140806034742.1438.25598.stgit@gir.stowe>
Patchwork-id: 87443
O-Subject: [RHEL7.1 PATCH 34/69] ACPI / hotplug / PCI: Use list_for_each_entry() for bus traversal
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: c6f0d5adc21e2d9623fe757f35a0a642a1d863c4

commit c6f0d5adc21e2d9623fe757f35a0a642a1d863c4
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Feb 13 21:13:58 2014 +0800

    ACPI / hotplug / PCI: Use list_for_each_entry() for bus traversal

    Replace list_for_each() + pci_bus_b() with list_for_each_entry().

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rjw@rjwysocki.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index c979fe7..6a440d4 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -450,7 +450,7 @@ static void cleanup_bridge(struct acpiphp_bridge *bridge)
  */
 static unsigned char acpiphp_max_busnr(struct pci_bus *bus)
 {
- struct list_head *tmp;
+ struct pci_bus *tmp;
  unsigned char max, n;
 
  /*
@@ -463,8 +463,8 @@ static unsigned char acpiphp_max_busnr(struct pci_bus *bus)
   */
  max = bus->busn_res.start;
 
- list_for_each(tmp, &bus->children) {
-  n = pci_bus_max_busnr(pci_bus_b(tmp));
+ list_for_each_entry(tmp, &bus->children, node) {
+  n = pci_bus_max_busnr(tmp);
   if (n > max)
    max = n;
  }
-- 
1.7.1